nodejs/week2/meal-sharing/src/backend/api/meals-router.js

```diff
@@ -0,0 +1,61 @@
+const express = require("express");
+const router = express.Router();
+const meals = require("./../data/meals");
+const notANumber = { message: "Not an integer" };
+const errorMessage = { message: "Something wrong" };
+
+router.get("/:id", async(request, response) => {
+    const neededId = Number(request.params.id);
+    if (!Number.isInteger(needdedId)) {
+        return response.status(400).json(notANumber);
+    } else {
+        let newObj = {};
+        meals.find((element) => {
```

Comment:
You are using find for iteration which is not a good practice. You could just send back what find returns in the response:
response.send(meals.find(...))

Reply:
Thank you. I changed it.